scheduler/scheduler-server/src/test/java/org/ow2/proactive/scheduler/core/CheckEligibleTaskDescriptorScriptTest.java

```diff
@@ -0,0 +1,245 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.ow2.proactive.scheduler.common.SchedulerConstants;
+import org.ow2.proactive.scheduler.common.task.ForkEnvironment;
+import org.ow2.proactive.scheduler.common.task.flow.FlowScript;
+import org.ow2.proactive.scheduler.descriptor.EligibleTaskDescriptor;
+import org.ow2.proactive.scheduler.task.containers.ScriptExecutableContainer;
+import org.ow2.proactive.scheduler.task.internal.InternalScriptTask;
+import org.ow2.proactive.scheduler.task.internal.InternalTask;
+import org.ow2.proactive.scripting.InvalidScriptException;
+import org.ow2.proactive.scripting.Script;
+import org.ow2.proactive.scripting.SimpleScript;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 23/06/2017
+ */
+public class CheckEligibleTaskDescriptorScriptTest {
+
+    private EligibleTaskDescriptor etd;
+
+    private InternalTask it;
+
+    private InternalScriptTask ist;
+
+    private ScriptExecutableContainer sec;
+
+    private ForkEnvironment fe;
+
+    @Before
+    public void init() {
+        etd = mock(EligibleTaskDescriptor.class);
+        it = mock(InternalTask.class);
+        ist = mock(InternalScriptTask.class);
+        sec = mock(ScriptExecutableContainer.class);
+        fe = mock(ForkEnvironment.class);
+
+        Mockito.when(etd.getInternal()).thenReturn(it);
+        Mockito.when(ist.getExecutableContainer()).thenReturn(sec);
+        Mockito.when(it.getForkEnvironment()).thenReturn(fe);
+        Mockito.when(ist.getForkEnvironment()).thenReturn(fe);
+    }
+
+    @Test
+    public void testAllScriptsContainAPIBinding() throws InvalidScriptException {
+        Script s = scriptWithApiBindingClient();
+        Script s1 = scriptWithApiBindingUser();
+        Script s2 = scriptWithApiBindingGlobal();
+        FlowScript fs = flowScriptWithApiBindingClient();
+        Mockito.when(it.getPreScript()).thenReturn(s1);
+        Mockito.when(it.getPostScript()).thenReturn(s2);
+        Mockito.when(sec.getScript()).thenReturn(s);
+        Mockito.when(fe.getEnvScript()).thenReturn(s);
+        Mockito.when(it.getFlowScript()).thenReturn(fs);
+        Boolean result = new CheckEligibleTaskDescriptorScript().isTaskContainsAPIBinding(etd);
+        assertEquals(true, result);
+    }
+
+    @Test
+    public void testAllScriptsNotContainAPIBinding() throws InvalidScriptException {
+        Script s = scriptWithoutApiBinding();
+        FlowScript fs = flowScriptWithoutApiBinding();
+        Mockito.when(it.getPreScript()).thenReturn(s);
+        Mockito.when(it.getPostScript()).thenReturn(s);
+        Mockito.when(sec.getScript()).thenReturn(s);
+        Mockito.when(fe.getEnvScript()).thenReturn(s);
+        Mockito.when(it.getFlowScript()).thenReturn(fs);
+        Boolean result = new CheckEligibleTaskDescriptorScript().isTaskContainsAPIBinding(etd);
+        assertEquals(false, result);
```

Comment:
use assertFalse(result) or assertThat(result, is(false)) instead of assertEquals which is not relevant in this case.